vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-integration-tests/src/main/java/com/vaadin/flow/component/spreadsheet/tests/fixtures/SimpleCustomEditorFixture.java

```diff
@@ -1,57 +1,193 @@
+                    componentType.getHeaderText());
+        }
+        spreadsheet.setSpreadsheetComponentFactory(new CustomEditorFactory());
+    }
+}
+
+class CustomEditorFactory implements SpreadsheetComponentFactory {
+
+    private TextField textField;
+
+    private TextArea textArea;
+
+    private Checkbox checkbox;
+
+    private DatePicker datePicker;
+
+    private ComboBox<String> comboBox;
+
+    @Override
+    public Component getCustomComponentForCell(Cell cell, int rowIndex,
+            int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
+        return null;
+    }
+
+    @Override
+    public Component getCustomEditorForCell(Cell cell, final int rowIndex,
+            final int columnIndex, final Spreadsheet spreadsheet, Sheet sheet) {
+        EditorType editorType = EditorType.getEditorTypeByIndex(rowIndex,
+                columnIndex);
+        if (editorType == null) {
+            return null;
+        }
+        return getCustomEditor(editorType, rowIndex, columnIndex, spreadsheet);
+    }
 
-        // spreadsheet.setSpreadsheetComponentFactory(new ComponentFactory());
-
-        List<Cell> cellsToRefresh = new ArrayList<Cell>();
-        cellsToRefresh.add(spreadsheet.createCell(0, 0, 1));
-
-        // this has no effect. sheet must be scrolled to "redisplay" the cell.
-        spreadsheet.refreshCells(cellsToRefresh);
-    }
-
-    // TODO Re-enable when SpreadsheetComponentFactory is done
-    // @SuppressWarnings("serial")
-    // public static class ComponentFactory
-    // implements SpreadsheetComponentFactory {
-    //
-    // private final TextField textField = new TextField("");
-    //
-    // @Override
-    // public Component getCustomComponentForCell(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
-    // return null;
-    // }
-    //
-    // @Override
-    // public Component getCustomEditorForCell(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet) {
-    // if (cell != null) {
-    // // textField.focus();
-    // return textField;
-    // }
-    // return null;
-    // }
-    //
-    // @Override
-    // public void onCustomEditorDisplayed(Cell cell, int rowIndex,
-    // int columnIndex, Spreadsheet spreadsheet, Sheet sheet,
-    // Component customEditor) {
-    // if (cell != null) {
-    // String cellValue = spreadsheet.getCellValue(cell);
-    // textField.setValue(cellValue);
-    // }
-    // }
-    // }
+    @Override
+    public void onCustomEditorDisplayed(Cell cell, int rowIndex,
+            int columnIndex, Spreadsheet spreadsheet, Sheet sheet,
+            Component customEditor) {
+        if (cell == null) {
+            return;
+        }
+        EditorType editorType = EditorType.getEditorTypeByIndex(rowIndex,
+                columnIndex);
+        if (editorType == null) {
+            return;
+        }
+        Component editor = getCustomEditor(editorType, rowIndex, columnIndex,
```

Comment:
`onCustomEditorDisplayed` gets the custom editor as an argument (`Component customEditor`) so it shouldn't be necessary to use `getCustomEditor` here